solidity/test/integration/WalleCreation.test.ts

```diff
@@ -0,0 +1,119 @@
+}
+
+const describeFn =
+  process.env.NODE_ENV === "integration-test" ? describe : describe.skip
+
+describeFn("Integration Test - Wallet Creation", async () => {
+  let bridge: Bridge
+  let walletRegistry: WalletRegistry
+  let randomBeacon: FakeContract<IRandomBeacon>
+  let governance: SignerWithAddress
+
+  const dkgResultChallengePeriodLength = 10
+
+  // TODO: Generate a random public key
+  const walletPublicKey = ecdsaWalletTestData.publicKey
+  const { walletID } = ecdsaWalletTestData
+  const walletPubKeyHash = ecdsaWalletTestData.pubKeyHash160
+
+  before(async () => {
+    ;({ governance, bridge, walletRegistry, randomBeacon } =
+      await waffle.loadFixture(fixture))
+
+    // Update only the parameters that are crucial for this test.
+    await updateWalletRegistryDkgResultChallengePeriodLength(
+      walletRegistry,
+      governance,
+      dkgResultChallengePeriodLength
+    )
+  })
+
+  describe("new wallet creation (happy path)", async () => {
+    let requestNewWalletTx: ContractTransaction
+    let walletRegistrationTx: ContractTransaction
+
+    before(async () => {
+      expect(await bridge.activeWalletPubKeyHash()).to.be.equal(
+        ethers.constants.AddressZero
+      )
+
+      requestNewWalletTx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+      const startBlock = requestNewWalletTx.blockNumber
+
+      const relayEntry: BigNumberish = await produceRelayEntry(
+        walletRegistry,
+        randomBeacon
+      )
+
+      ;({ approveDkgResultTx: walletRegistrationTx } =
+        await produceEcdsaDkgResult(
+          walletRegistry,
+          walletPublicKey,
+          relayEntry,
+          startBlock
+        ))
+
+      await walletRegistrationTx.wait()
+    })
+
+    it("should register a new wallet in the WalletRegistry", async () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      expect(await walletRegistry.isWalletRegistered(walletID)).to.be.true
+    })
+
+    it("should register a new wallet details in the Bridge", async () => {
+      const storedWallet = await bridge.wallets(walletPubKeyHash)
+
+      expect(storedWallet.ecdsaWalletID).to.be.equal(
+        ecdsaWalletTestData.walletID
+      )
+
+      expect(storedWallet.state).to.be.equal(1)
+
+      expect(storedWallet.createdAt).to.be.equal(
+        (
+          await ethers.provider.getBlock(
+            (
+              await walletRegistrationTx.wait()
+            ).blockNumber
+          )
+        ).timestamp
+      )
+    })
+
+    it("should register a new wallet as active in the Bridge", async () => {
+      expect(await bridge.activeWalletPubKeyHash()).to.be.equal(
+        walletPubKeyHash
+      )
+    })
+
+    it("should consume around 93 000 gas for Bridge.requestNewWallet transaction", async () => {
+      await assertGasUsed(requestNewWalletTx, 93_000)
+    })
+
+    // TODO: Should we also validate gas used by Random Beacon and ECDSA Wallet Registry transactions?
```

Comment:
At a minimum, the one approving result - it's calling the `Bridge` in a callback.

Reply:
04f418bc529cc75f870c133e53dd0b5ac99a9b40